reader.go

```diff
@@ -183,6 +257,43 @@ func (a *reader) Read(p []byte) (n int, err error) {
 	return n, nil
 }
 
+func (a *reader) Seek(offset int64, whence int) (res int64, err error) {
+	if a.seeker == nil {
+		err = fmt.Errorf("seeker not found")
+	} else {
+		//Make sure the async routine is closed
+		select {
+		case <-a.exited:
+		case a.exit <- struct{}{}:
+			<-a.exited
+		}
+		if whence == SeekCurrent {
+			a.fill()
+			//If need to seek based on current position, take into consideration the bytes we read but the consumer
```

Comment:
This seems to me like it should be a `for` loop than ensures that all buffers are drained. Right now, as far as I can tell it will only add one block.

Reply:
Well there is  a `for` loop right under this, which drains the ready channel, I might miss something, but I believe that's the only one that needs to be drained and taken into consideration.